gpsd fails to comipile with build error related to missing curses library

-------------------------------

Maintainer: @psidhu @p-wassi 
Environment: OMAP, BeagleBone, LEDE-17.01

Description:
gpsd does not compile under lede-17.01 branch, on OMAP (other arches not tested).
the linking step for cgps fails with undefined references to functions that should be provided by a curses library. Full error output is in [this gist](https://gist.github.com/mrnuke-adaptrum/cce7c6ec7dc433bcd8aaf36d580929da)

This also fixed cross-compilation on macOS for OMAP, but inadvertently by the look of it. I patched it a different way, but my problem was with SConstruct in that it checks sys.platform() against darwin and if detected assumes ncurses will be "libcurses" as opposed to "libncurses" even though it's cross compiling for OMAP.  

I was getting this linking error: -

> lede-source-ts/staging_dir/toolchain-mips_24kc_gcc-5.4.0_musl-1.1.16/lib/gcc/mips-openwrt-linux-musl/5.4.0/../../../../mips-openwrt-linux-musl/bin/ld: cannot find -lcurses
> 

The change you made is forcing "ncurseslibs" variable via the configuration script prior to the platform check, which on my macOS environment is currently using brew which results in "libncursesw", so I get this: -

> /usr/local/Cellar/ncurses/6.0/bin/ncurses6-config --libs --flags
> -L/usr/local/Cellar/ncurses/6.0/lib -lncursesw
> -D_DARWIN_C_SOURCE -I/usr/local/Cellar/ncurses/6.0/include/ncursesw -I/usr/local/Cellar/ncurses/6.0/include

FYI, my version of macOS (10.12.4) looks to include ncurses 5.4: -

> /usr/bin/ncurses5.4-config --libs --flags
> -lncurses
> -I/usr/include/ncurses
> 

So the original assumption in SConstruct that the macOS is "libcurses" might be wrong or dated... I don't know enough to comment further, but your change fixed my issue! 👍 

Just letting you know :)

--- a/utils/gpsd/patches/0002-ncurses6_detection.patch
+++ b/utils/gpsd/patches/0002-ncurses6_detection.patch
@@ -1,0 +1,15 @@
+Index: gpsd-3.16/SConstruct
+===================================================================
+--- gpsd-3.16.orig/SConstruct
++++ gpsd-3.16/SConstruct
+@@ -543,6 +543,10 @@ else:
+             ncurseslibs = pkg_config('ncurses')
+ 	    if config.CheckPKG('tinfo'):
+ 		ncurseslibs += pkg_config('tinfo')
++        elif WhereIs('ncurses6-config'):
++            ncurseslibs = ['!ncurses6-config --libs --cflags']
++        elif WhereIs('ncursesw6-config'):
++            ncurseslibs = ['!ncursesw6-config --libs --cflags']
+         elif WhereIs('ncurses5-config'):
+             ncurseslibs = ['!ncurses5-config --libs --cflags']
+         elif WhereIs('ncursesw5-config'):

